CONFIGS.md
# Configs

The bot uses & generates various configuration files. This is the documentation ~~because I'm 100%
bound to forget what they mean~~ because I can't add comments to them themselves (json file moment)

---

## ./

### config.json

Main config file

```JSONC
{   
    // Prefix to interpret commands with
    // either list[str] or str 
    "prefix": ["r.", "r!"],
    // status used in initialization, where {} becomes the prefix
    // used in cogs/events/stati.py
    "default_status": "{}help",
    // PYTZ-compatible timezone
    // see https://gist.github.com/heyalexej/8bf688fd67d7199be4a1682b3eec7568
    "preferred_timezone": "EST",
    // link to a repo
    // used in cogs/commands/info.py
    "github": "https://github.com/ernieIzde8ski/ratbot",
    // invite to a guild
    // used in cogs/commands/info.py
    "invite": "https://discord.gg/3gfP2kYPp4",
    // id of the above guild
    "main_guild": 488475203303768065,
    // Channel IDs to send various messages to
    "channels": {
        "BMs": 762166605458964510,
        "DMs": 715297562613121084,
        "Status": 708882977202896957,
        "Guilds": 841863106996338699
    }
}
```

### enabled_extensions.json

List of extensions the bot runs in initialization. These can be found in the cogs directory.

These extensions contain base features:

```JSON
["cogs.commands.cogs", "cogs.commands.info", "cogs.events.guilds", "cogs.events.log", "cogs.events.stati", "cogs.on_message.dm_replies", "cogs.on_message.dms"]
```

Other extensions are dependent on these:

```JSON
["cogs.commands.randomized", "cogs.commands.settings"]
```

## ./utils/JSON/

### borg.json

List sampled in cogs/commands/star_trek.py

### petrosian.json

Copypasta used in its respectively named cog.

## ./data/

### banning.json

Dictionary of servers that have enabled random bans.

```JSON
{"488475203303768065": 2e-05}
```

### bible.json

Dictionary of preferred versions

```JSON
{"302956027656011776": "kjv"}
```

### blocked.json

List of users blocked from the bot

```JSON
[380541859526737942]
```

### bm.json

List of inputs already evaluated in the past in randomized cog

```JSON
["Armenium", "Ernie", "Glinky", "Gilnky"]
```

### emoji.json

2 emojis used in sending a message on startup

```JSON
["<:online:708885917133176932>", "<:offline:708886391672537139>"]
```

### lmfao.json

Emoji used in reacting to a message

```JSON
"<:hhahh:857336426819747890>"
```

### prefixes.json

Guild-specific command prefixes

```JSON
{"488475203303768065": "r;", "387729240008818689": "r;"}
```

### russian.json

The first 10,000 verses or so of a Russian bible. Generated by utils/russian.py.

```JSON
["В начале сотворил Бог небо и землю.", "Земля же была безвидна и пуста, и тьма над бездною, и Дух Божий носился над водою.", "..."]
```

### songs.json

A dictionary of songs.

```JSON
{
    "AjZrV4wbdnQ": "Aleksandr Marshal & Valeria Kurnushkina — Katyusha (Катюша)",
    "j9BcQcFVcRM": "Bachman Turner Overdrive — You Ain't Seen Nothing Yet",
    "GUHQmQfsBYY": "Bathory — A Fine Day to Die",
    "VI6elvIoRK0": "Bathory — Ring of Gold",
}
```

### tenor_guilds.json

A list of guilds with tenor-deleting enabled.

```JSON
[488475203303768065]
```

### timekeeping.json

A dictionary of user timezones.

```JSON
{"302956027656011776": "America/Los_Angeles"}
```

### trolls.json

A list of troll emojis to react with.

```JSON
["<:untroll:857332422584172566>", "<:trort:857330947174957057>", "<:trollsmoothest:857330981702467644>", "<:trollsad:857332238109376553>", "<:trollexcited:857331745119535144>", "<:trollcraft:857328836329078815>", "<:troll:857331191034937364>", "<:trolfl:857328783345319936>", "<:tro:857330854538510346>", "<:redditroll:857332382343102514>", "<:celeste_granny:830955083692113920>", "<:eltroll:859149163145330708>"]
```

### xkcd.json

Cache of XKCD files. Generated by utils/xkcd_cache_generator.js, and validated by
utils/xkcd_cache_checker.js. Used in cogs/commands/xkcd.py.


configs/config.py
class Config:
    def __init__(self):
        self.prefix = ["r.", "r!"]  # prefix that bot uses; must be list
        # relevant to cogs.on_message.respuestas
        self.guild_opt_in = [488475203303768065, 526207286067068928, 534875827398246451]
        # guilds which have opted into tenor slaying
        self.SixPP_guilds = [488475203303768065]  # guilds with random bans
        self.SixPP_Chance = 0.00002  # chance of random bans
        # this value defaults to 0.00002,
        # which gives you about a 0.1% chance of getting banned within 500 messages,
        # or 9.5% chance within 5000 messages
        # P(banned in N messages) = 1 - (1-SixPP_Chance) ^ N
        self.channels = Channels()  # the channels lol
        self.adminname = "ernie"  # bot admin name - must be lowercase
        self.corrections = {  # cogs.on_message.corrections
            "bano": "Baño",
            "senor": "Señor",
            "senora": "Señora",
            "jalapeno": "Jalapeño",
            "canada": "Cañada",
            "canadian": "Cañadian",
            "retard": "Ratard",
            "ExampleSlur": "Armenium"
        }
        # songs referenced in cogs.core.[help/info] & cogs.randomized via youtube url
        self.songs = {
            "AjZrV4wbdnQ": "Aleksandr Marshal & Valeria Kurnushkina — Katyusha (Катюша)",
            "j9BcQcFVcRM": "Bachman Turner Overdrive — You Ain't Seen Nothing Yet",
            "GUHQmQfsBYY": "Bathory — A Fine Day to Die",
            "VI6elvIoRK0": "Bathory — Ring of Gold",
            "YEMEAxlYL04": "The Beatles — While My Guitar Gently Weeps",
            "mJag19WoAe0": "The Beatles — Maxwell's Silver Hammer",
            "j_JaDDcyIIU": "The Beatles — Yellow Submarine",
            "hCuMWrfXG4E": "Billy Joel — Uptown Girl",
            "8nWLZaijTbk": "Blind Guardian — Another Holy War",
            "2yGfqgESJxY": "Blind Guardian — Harvest of Sorrow",
            "M0TcB5lxfuY": "Blind Guardian — Nightfall",
            "bRLML36HnzU": "Bobby Pickett — Monster Mash",
            "oe2hdbft5-U": "Bob Marley & The Wailers — I Shot The Sheriff (1973)",
            "W1LsRShUPtY": "Bob Seger — Old Time Rock & Roll",
            "No08DrgSy0Y": "Cool and New Music Team — Dr. Pepper",
            "tKJwvQfraY8": "Creedence Clearwater Revival — Bad Moon Rising",
            "RE70Pv5yRSw": "Dark Oath — Brother's Fall",
            "3uX9qK_f4XU": "Dark Oath — Brother's Fall",
            "jtgA0jvhp2A": "Dubioza kolektiv — No Escape (from Balkan)",
            "uGrb-6Vb9Bw": "Dünedain — Por los siglos de los siglos",
            "WZ4beMU9wBU": "Dünedain — Memento Mori",
            "aJfgHSUlr-s": "The Coasters — Yakety Yak",
            "4v8KEbQA8kw": "The Eagles — Take It Easy",
            "alQei8zVMyM": "Elina Ohanessian — Kroonk",
            "Jxj9w_7JasU": "Eluveitie — The Nameless",
            "TRUZ0gB_xKY": "Elvenking — Towards the Shores",
            "fYTgJVDYWNs": "Elvenking — Twilight of Magic",
            "2KxwqyM7k2s": "Elvenking — The Winter Wake (Acoustic Version)",
            "4gNR7UDSLXo": "Emmanuel — Esa Triste Guitarra",
            "O0PV0M6-j9w": "Eric Clapton — Layla",
            "VVqXLXMch9g": "Eric Clapton — Tears in Heaven",
            "IvVe-uHRD-U": "ernieIzde8ski — smells like nirvana but only the gargling",
            "hTkW8DLVpwA": "The Flaming Lips — Yoshimi Battles The Pink Robots Pt. 1",
            "OZuW6BH_Vak": "Heart — Crazy On You",
            "eLoMej34zvA": "Heart — Dreamboat Annie",
            "vUl6hPCYAvc": "Helloween — Eagle Fly Free",
            "9XHQpoG5E0w": "Helloween — Rise and Fall",
            "bCNHl2wMPQw": "Helloween — Keeper of the Seven Keys",
            "Es5mh7pBeec": "Homestuck: [S] Collide Track 4 — Heir of Grief",
            "q-822c9-d6I": "Iron Maiden — The Evil That Men Do",
            "s2WgQtPUsuQ": "Iron Maiden — Seventh Son of a Seventh Son",
            "7-iRf9AWoyE": "Iron Maiden — The Number of the Beast",
            "VkuY33xb6v8": "The Kentucky Colonels — There Ain't Nobody Gonna Miss Me When I'm Gone",
            "S7Jw_v3F_Q0": "The Kingston Trio — M.T.A.",
            "yBuub4Xe1mw": "Led Zeppelin — Black Dog",
            "eYNMcolpHEM": "Meat Loaf — Wasted Youth",
            "W-0qx0yf_Hg": "Mormon Tabernacle Choir — Old Time Religion",
            "BJhMjuza_1A": "Mormon Tabernacle Choir — Rock-a-My Soul in the Bosom of Abraham",
            "MYKbQVw80mI": "Petra — Seen and Not Heard",
            "lrNnMNrKqQc": "Rhapsody — Symphony of the Enchanted Lands",
            "TJO9oEW_fa8": "Rhapsody — When Demons Awake",
            "FqnhEm8KZFU": "Rhapsody — The Village of Dwarves",
            "aXvG_Lx0Kp4": "Selo i Ludy — I Want To Break Free",
            "ha0icvcByDs": "Styx — The Greater Good",
            "i-mWU2JFvUU": "Twilight Force — Blade of Immortal Steel",
            "Eq7-DsMhLaA": "Twilight Force — Night of Winterlight",
            "B7joqHpvSXo": "Twilight Force — Heroes of Mighty Magic",
            "YZVJb1dyiV8": "Wise Guys — Ans Ende der Welt",
            "zjKLdZcG8X8": "Wise Guys — Das Sägewerk Bad Sebeberg",
            "MoN3zUJb6tA": "Wise Guys — Ruf' doch mal an '06",
            "j_nuOyxMrMQ": "The Wolfe Tones — Come Out Ye Black & Tans"
        }


class Channels:
    def __init__(self):
        self.status = 708882977202896957  # channel where startup is logged
        self.log = 715297562613121084  # channel where miscellaneous actions [i.e. dm messages to rat] are logged
        self.bm = 762166605458964510  # channel where based meter is stored
        self.guilds = 841863106996338699  # channel where guild updates are logged

    def _get_channels(self, bot):
        self.status = bot.get_channel(self.status)
        self.log = bot.get_channel(self.log)
        self.bm = bot.get_channel(self.bm)
        self.guilds = bot.get_channel(self.guilds)
